src/file_dialog.rs

```diff
@@ -1331,8 +1346,12 @@ impl FileDialog {
 
     /// Updates the view when the user currently wants to text edit the current path.
     fn ui_update_path_edit(&mut self, ui: &mut egui::Ui, width: f32, edit_button_size: egui::Vec2) {
-        let desired_width: f32 =
-            width - edit_button_size.x - ui.style().spacing.item_spacing.x * 3.0;
+        let desired_width: f32 = ui
+            .style()
+            .spacing
+            .item_spacing
+            .x
+            .mul_add(-3.0, width - edit_button_size.x);
```

Comment:
I think this also makes reading and understanding the code harder than it needs to be.

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#suboptimal_flops

See here.
Yes not that easy to read but better for performance...

Reply:
Ye, I know, but the performance benefit is so extremely small that I'm not sure it justifies making the code quite harder to maintain and understand.

I think the lint belongs to nursery, right? Since you have activated it now, I would say let's leave it like that for now. Putting a nolint there is just as ugly ^^

Reply:
We could deactivate that specific Lint...

Reply:
In some situations the lint is helpful, especially if the calculations are not too complicated. In reality, people often don't pay attention to things like this, so I think lint is useful overall.

We'll leave it on for now and stick to the way you've done it here.

I'll do another review of your changes tomorrow and then merge the PR. I don't have time for that anymore today.